File: ios/MendixNative/RuntimeInfoProvider/RuntimeInfo.m
#import "RuntimeInfo.h"

@implementation RuntimeInfo

- (instancetype) initWithVersion:(NSString *)version cacheburst:(NSString *)cacheburst nativeBinaryVersion:(long)nativeBinaryVersion packagerPort:(long)packagerPort {
  self = [super init];
  if (self) {
    _version = version;
    _nativeBinaryVersion = nativeBinaryVersion;
    _packagerPort = packagerPort;
    _cacheburst = cacheburst;
  }
  return self;
}

@end

File: ios/Modules/RuntimeInfoProvider/RuntimeInfo.swift
import Foundation

public struct RuntimeInfo {
    
    // MARK: - Properties
    public let version: String
    public let cacheburst: String
    public let nativeBinaryVersion: Int
    public let packagerPort: Int
    
    static func initWith(_ dictionary: [String: Any]) -> Self {
        let version = dictionary["version"] as? String ?? ""
        let cacheburst = dictionary["cachebust"] as? String ?? ""
        let nativeBinaryVersion = dictionary["nativeBinaryVersion"] as? Int ?? 0
        let packagerPort = dictionary["packagerPort"] as? Int ?? 0
        return RuntimeInfo(
            version: version,
            cacheburst: cacheburst,
            nativeBinaryVersion: nativeBinaryVersion,
            packagerPort: packagerPort
        )
    }
}
